### PR TITLE
Do not automatically add UID 0 to proxy

### PR DESCRIPTION
**Pull request by rshriram**
This seems like vestigial code that causes lots of IPtables woes.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ x] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[x ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: tools/istio-iptables/pkg/cmd/root.go [hunk 1]
```diff
@@ -109,15 +109,12 @@ func constructConfig() *config.Config {
...
-		// Default to the UID of ENVOY_USER and root
+		// Default to the UID of ENVOY_USER
 		if err != nil {
 			userID = constants.DefaultProxyUID
 		} else {
 			userID = usr.Uid
 		}
-		// If ENVOY_UID is not explicitly defined (as it would be in k8s env), we add root to the list
-		// for the CA agent.
-		cfg.ProxyUID = userID + ",0"
```
Comment (howardjohn) [inline tools/istio-iptables/pkg/cmd/root.go:120]: don't we need this? its only hitting this code path if its not set, meaning its not set on VMs currently?

Comment (rshriram) [reply tools/istio-iptables/pkg/cmd/root.go:120]: Oh sorry.. i was editing directly from GH!

**Comment by rlenglet** [comment]
This is for VM support only, IIRC. Removing this should break VM support?

**Comment by howardjohn** [comment]
We had it to avoid capture of the node-agent component (per comment, at
least), which no longer exists. I think *having* this actually breaks VMs
if the app runs as root? That is my understanding at least

**Pull request merged**